tests/CMakeLists.txt

```diff
@@ -64,8 +64,9 @@ set(API_TEST_DEPS oqs ${LIBM})
 add_executable(example_kem example_kem.c)
 target_link_libraries(example_kem PRIVATE ${API_TEST_DEPS})
 
-add_executable(kat_kem kat_kem.c)
-target_link_libraries(kat_kem PRIVATE ${API_TEST_DEPS})
+# KAT KEM needs to call the internal SHA3 functions directly, hence the extra dependencies
+add_executable(kat_kem kat_kem.c ${COMMON_OBJS})
```

Comment:
Hmm -- this line makes me wonder: If one adds all COMMON_OBJS _and_ links against `liboqs` (line below, API_TEST_DEPS) which symbols "win"? The ones from the objects list or the ones from the library?

Reply:
> Hmm -- this line makes me wonder: If one adds all COMMON_OBJS _and_ links against `liboqs` (line below, API_TEST_DEPS) which symbols "win"? The ones from the objects list or the ones from the library?

Maybe the "winner" is picked differently on Mac vs Linux...

Reply:
What about this theory: `oqs_randombytes_algorithm` is a static function pointer in an object file; Linux keeps using that throughout the code after it has been initialized by the call in `kat_kem.c`. MacOS in turn "overloads" it with a new one when loading the library for all further invocations of library functions. I don't quite understand why the lib doesn't get loaded right at the moment `OQS_randombytes_switch_algorithm` gets called, though.... So, just a wild theory, still no idea how to fix.